### ISSUE TITLE
500 Error code for a Bad XML body 

### ISSUE THREAD

**Issue by akshaybheda** [DigDes/SoapCore#1021]
I was trying to get my service out to production, it gets 100 req per second per instance, i am replacing the old WCF with SOAPCore, I started to see these error. I tried to figure out whats going on but I don't log req headers or body in the logs, 

this is the stacktrace i found for the exception

```
System.Xml.XmlException:
   at System.Xml.XmlExceptionHelper.ThrowXmlException (System.Private.DataContractSerialization, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
   at System.Xml.XmlUTF8TextReader.Read (System.Private.DataContractSerialization, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
   at System.Xml.XmlBaseReader.ReadEndElement (System.Private.DataContractSerialization, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
   at System.ServiceModel.Channels.Message.ReadFromBodyContentsToEnd (System.Private.ServiceModel, Version=4.8.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
   at System.ServiceModel.Channels.ReceivedMessage.OnWriteBodyContents (System.Private.ServiceModel, Version=4.8.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
   at System.ServiceModel.Channels.Message.OnWriteMessage (System.Private.ServiceModel, Version=4.8.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
   at System.ServiceModel.Channels.Message.OnCreateBufferedCopy (System.Private.ServiceModel, Version=4.8.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
   at SoapCore.HeadersHelper.GetSoapAction (SoapCore, Version=1.1.0.41, Culture=neutral, PublicKeyToken=092c4b76488e2b26: /_/src/SoapCore/HeadersHelper.cs:21)
   at SoapCore.SoapEndpointMiddleware`1+<ProcessMessage>d__17.MoveNext (SoapCore, Version=1.1.0.41, Culture=neutral, PublicKeyToken=092c4b76488e2b26: /_/src/SoapCore/SoapEndpointMiddleware.cs:455)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at SoapCore.SoapEndpointMiddleware`1+<ProcessOperation>d__14.MoveNext (SoapCore, Version=1.1.0.41, Culture=neutral, PublicKeyToken=092c4b76488e2b26: /_/src/SoapCore/SoapEndpointMiddleware.cs:304)
```

Invalid characters at the end of request body throwing 500 where as it should be 400

**Comment by akshaybheda** [comment]
and there's some performance issue as well while reading the body for the soap request, i got the memory dump, and this is the img 
![image](https://github.com/DigDes/SoapCore/assets/22575879/aa048d28-98e2-4476-af05-4c743f32eabe)

**Comment by andersjonsson** [comment]
Fixed by #1025
